src/main/kotlin/com/manimdsl/runtime/ExecValue.kt

```diff
@@ -16,19 +16,46 @@ sealed class ExecValue {
 
     /** '+' **/
     operator fun plus(other: ExecValue): ExecValue = when (this) {
-        is DoubleValue -> if (other is DoubleValue) DoubleValue(this.value + other.value) else throwTypeError()
+        is DoubleValue -> when (other) {
+            is CharValue -> DoubleValue((other.value.toDouble() + this.value))
+            is DoubleValue -> DoubleValue((this.value + other.value))
+            else -> throwTypeError()
+        }
+        is CharValue -> when (other) {
+            is CharValue -> DoubleValue((this.value.toDouble() + other.value.toDouble()))
+            is DoubleValue -> DoubleValue((this.value.toDouble() + other.value))
+            else -> throwTypeError()
+        }
         else -> throwTypeError()
     }
 
     /** '-' **/
     operator fun minus(other: ExecValue): ExecValue = when (this) {
-        is DoubleValue -> if (other is DoubleValue) DoubleValue(this.value - other.value) else throwTypeError()
+        is DoubleValue -> when (other) {
```

Comment:
Could this not be:

`DoubleValue((other.value.toString().toDouble() - this.value.toString().toDouble()))` 

Do we ever hit the throwTypeError? We shouldn't right because it must be semantically correct?

Same for the other operators, if this would work

Reply:
This will not work for chars as they have their codes so 'a' = 97 and "a".toDouble() would throw an error

Reply:
Ah yeah true, what about having each PrimtiveExecValue having a field which is the double representation (or it's own toDouble function) and then there's no need for when clauses etc

Reply:
And as before, if it's semantically right, this will never try to do a stack plus or minus operation for example

Reply:
I don't think we want to be able to add strings to doubles. I just wanted it to be limited to chars.

Reply:
Makes sense, in which case cast it to an interface which has a toDouble function, so that characters and doubles can be added but nothing else

Reply:
Also more importantly, that would be stopped at the semantic level, and wouldn't reach the executor